### ISSUE TITLE
How can I perform data training on a custom binary segmentation dataset?

### ISSUE THREAD

**Issue by MrLangZe** [qubvel-org/segmentation_models.pytorch#1056]
I tried to build my own dataset for training by following the code example at https://github.com/qubvel-org/segmentation_models.pytorch/blob/main/examples/cars%20segmentation%20(camvid).ipynb, but failed. How can I replace my own binary segmentation dataset in the code?

**Comment by qubvel** [comment]
Hey @MrLangZe, sorry to hear you get some issues. You just need to build your own dataset class which will return image and binary mask, the rest should be the same.
The mask should contain 0 and 1 values and to be of shape (batch_size, 1, height, width) -  with the same height and width as image.

**Comment by MrLangZe** [comment]
Thank you, I have another question, can I use smp.metrics.accuracy() to get the MPA (Mean Pixel Average) indicator results? I tried to set a class_weights list, but the result was not as expected.

**Comment by MrLangZe** [comment]
> Thank you, I have another question, can I use smp.metrics.accuracy() to get the MPA (Mean Pixel Average) indicator results? I tried to set a class_weights list, but the result was not as expected.

Hello, it seems that in this code segment, I tried to implement the metric MPA using the following formula: MPA = 1/classes *sum (TP / (TP + FN + TN + FP)), but I encountered difficulties due to my unfamiliarity with the interface. I wonder if you could give me some advice, thanks.
